pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -658,9 +665,26 @@ public QueryContext build() {
       queryContext._isUnsafeTrim =
           !queryContext.isSameOrderAndGroupByColumns(queryContext) || queryContext.getHavingFilter() != null;
 
+      // Pre-calculate group by trim size
+      queryContext._groupByTrimSize = getGroupByTrimSize(queryContext);
```

Comment:
These applies to group-by queries only. We can add a if check and calculate them only for group-by queries